artifacts/definitions/Windows/Registry/AppCompatCache.yaml

```diff
@@ -1,22 +1,87 @@
 name: Windows.Registry.AppCompatCache
+author: Matt Green - @mgreen27
 description: |
-  Parses the system's app compatibility cache.
+  This artifact parses AppCompatCache (shimcache) from target hives. 
+  
+  AppCompatCache, also known as Shimcache, is a component of the Application 
+  Compatibility Database, which was created by Microsoft and used by the Windows 
+  operating system to identify application compatibility issues. This helps 
+  developers troubleshoot legacy functions and contains data related to Windows 
+  features.
+  
+  Note: the appcompatcache plugin does not currently support execution flag in 
+  Windows 7 and 8/8.1 Systems.
+
+reference:
+  - https://www.mandiant.com/resources/caching-out-the-val
 
 parameters:
   - name: AppCompatCacheKey
     default: HKEY_LOCAL_MACHINE/System/ControlSet*/Control/Session Manager/AppCompatCache/AppCompatCache
+    
 
 precondition: SELECT OS From info() where OS = 'windows'
 
 sources:
   - query: |
-      LET AppCompatKeys = SELECT * FROM glob(
-          globs=AppCompatCacheKey, accessor='registry')
-      SELECT * FROM foreach(
-          row={
-              SELECT Filename, Data FROM read_file(
-                  filenames=AppCompatKeys.FullPath, accessor='registry')
-          }, query={
-              SELECT Filename AS Key, name as Name, epoch, time AS ModificationTime
-              FROM appcompatcache(value=Data)
-        }) WHERE epoch < 2000000000
+      -- first find all ControlSet Keys in scope
+      LET AppCompatKeys <= SELECT OSPath FROM glob(globs=AppCompatCacheKey, accessor='registry')
+      
+      -- when greater than one key we need to extract results and order later
+      LET results <= SELECT
+            ModificationTime,
+            Name as Path,
+            ControlSet,
+            Key
+          FROM foreach(
+              row={
+                  SELECT Filename, Data FROM read_file(
+                      filenames=AppCompatKeys.OSPath, accessor='registry')
+              }, query={
+                  SELECT Filename AS Key, Name, time AS ModificationTime,
+                    parse_string_with_regex(string=Filename,
```

Comment:
You dont really need this regex - it is simply OSPath[2]

Reply:
I think we probably need a test but we should look at a fix of the plugin first. We should parse 7/8/8.1 execution flags. Then have a test for various types